lib/src/deploy.rs

```diff
@@ -340,6 +347,64 @@ pub(crate) fn switch_origin_inplace(root: &Dir, imgref: &ImageReference) -> Resu
     Ok(newest_deployment)
 }
 
+pub fn get_kargs(repo: &ostree::Repo, fetched: &ImageState) -> Result<Vec<String>> {
+    let cancellable = gio::Cancellable::NONE;
+
+    // Get the running kernel commandline arguments
+    let kargs = ostree::KernelArgs::new();
+    ostree::KernelArgs::append_proc_cmdline(
```

Comment:
Note that the existing ostree logic actually picks them up from the booted deployment's bootloader entry - which is subtly different from `/proc/cmdline`.  One difference there is that some things are actually injected or interpreted by grub (e.g.) and we don't want to re-inject them into the kernel commandline.

IOW, the starting point should be from calling `ostree_deployment_get_bootconfig` on the booted deployment.  https://github.com/ostreedev/ostree/blob/3c25a93bac257acbbf01e3aa64cff29f3bb84f4f/src/ostree/ot-admin-builtin-deploy.c#L186